pkg/aws/aws.go

```diff
@@ -16,14 +17,28 @@ import (
 )
 
 func AnalyzeWaste(log *zap.SugaredLogger) {
-	sess := session.Must(session.NewSession())
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
 
-	region := viper.GetString(util.FlagRegion)
+	var region string
+
+	if viper.IsSet(util.FlagRegion) {
+		region = viper.GetString(util.FlagRegion)
+	} else if *sess.Config.Region != "" {
+		region = *sess.Config.Region
+	} else {
+		fmt.Println("No region provided or found in AWS config.")
```

Comment:
```suggestion
		log.Error("no region provided or found in AWS config.")
```
There's a whole logger to use.